src/client/utils/hooks/useLockBodyScroll.ts

```diff
@@ -0,0 +1,20 @@
+import { useLayoutEffect } from 'react'
+
+type UseLockBodyScrollConfig = { lock: boolean }
+
+function useLockBodyScroll({ lock }: UseLockBodyScrollConfig = { lock: true }) {
```

Comment:
Also can't you write like this?

```suggestion
function useLockBodyScroll({ lock = true }: UseLockBodyScrollConfig) {
```

doesn't it make more sense that the default is "false" btw?

Reply:
🤔 The idea of use `true` as the default value is because my initial idea for the api of this hook was:

- Create a hook that when used inside a component A would make A cause a block of scroll on the body element **while A is mounted**, which means everything would be back to normal as soon as **A is unmounted**

This makes sense if we start to think about the type of components that would probably use a hook like that: _Modal_s, for example. When a _Modal_ is opened it normally capture all of the user interactions so no scroll should (normally) be taken in place in the background. But as soon as the modal is resolved everything should be back to normal.

I like that approach very much but there's a corner case:

- Sometimes we want to keep a component like a _Modal_ mounted even though it's not being displayed..

In that case the initial API of this hook wouldn't work. That's why it also accepts a way to _control_ when scroll should be blocked or not. So if you want to use it blocking/unblocking based on mount/unmount just call it inside your component. However if you want to determine when it should block/unblock, just call it providing a proper value.

I'm sorry for the long post. Here, take a potato:
![image](https://user-images.githubusercontent.com/19200662/141002102-b03596e5-8661-4f56-9986-6812a2e3a75a.png)